sycl/doc/extensions/experimental/sycl_ext_oneapi_properties.asciidoc

```diff
@@ -228,10 +232,10 @@ template<typename V, typename O, typename=void> struct is_property_value_of;
 template<typename V> struct is_property_value<V, std::enable_if_t<(sizeof(V)>0)>> : is_property_key<V> {};
 template<typename V, typename O> struct is_property_value_of<V, O, std::enable_if_t<(sizeof(V)>0)>> : is_property_key_of<V,O> {};
 // Specialization for compile-time-constant properties
-template<typename V> struct is_property_value<V, std::void_t<typename V::key_t>> :
-  is_property_key<typename V::key_t> {};
-template<typename V, typename O> struct is_property_value_of<V, O, std::void_t<typename V::key_t>> :
-  is_property_key_of<typename V::key_t, O> {};
+template<typename V> struct is_property_value<V, std::void_t<key_from_value<V>>> : //key_from_value is exposition only
+  is_property_key<key_from_value<V>> {};
```

Comment:
Is `is_property_key` actual useful? This could instead just check that it is of type `property_value<T...>`. And if we remove `is_property_key` properties don't need to set that trait. And I'm only aware of useful usage of `is_property_key_of`. Why would a user care whether something is a property key rather than only whether it is a valid property for a certain usage?

Reply:
I think the constraint for `properties::has_property` and `properties::get_property` requires us to know whether a type is a property key.  How do we do that without the `is_property_key` trait?

Reply:
You're right the trait is useful for that and in that case we might as well expose it in case a user has some use for it. And we can still remove the need for the each property to set the trait in the implementation. We can instead e.g. check PropertyToKind.